app/jobs/importing/hud_zip/hmis_auto_migrate_job.rb

```diff
@@ -11,44 +11,45 @@ class HmisAutoMigrateJob < BaseJob
 
     attr_accessor :job
 
-    def initialize(upload_id:, data_source_id:, deidentified: false, project_whitelist: false)
+    def initialize(upload_id:, data_source_id:, deidentified: false, allowed_projects: false)
       @upload_id = upload_id
       @data_source_id = data_source_id
       @deidentified = deidentified
-      @project_whitelist = project_whitelist
+      @allowed_projects = allowed_projects
     end
 
     def before(job)
       self.job = job
     end
 
     def perform
-      lock_obtained = GrdaWarehouse::DataSource.with_advisory_lock(advisory_lock_name, timeout_seconds: 0) do
+      lock_obtained = GrdaWarehouse::DataSource.with_advisory_lock(advisory_lock_name, timeout_seconds: 60) do
         importer = Importers::HmisAutoMigrate::UploadedZip.new(
           data_source_id: @data_source_id,
           upload_id: @upload_id,
           deidentified: @deidentified,
-          allowed_projects: @project_whitelist,
+          allowed_projects: @allowed_projects,
         )
         importer.import!
       end
-      return if lock_obtained
 
       # when this exits, it will remove the current job from the queue, so add a new one to replace it
-      requeue_job
+      requeue_job unless lock_obtained
     end
 
     private def advisory_lock_name
-      "hud_import_auto_migrate_#{@data_source_id}"
+      GrdaWarehouse::DataSource.import_advisory_lock_name(data_source_id)
+    end
+
+    private def data_source_id
+      job.payload_object.instance_variable_get(:@data_source_id)
```

Comment:
sorry I'm missing something. Why isn't this just @data_source_id?

Reply:
nope, I think you are correct, I just blindly moved that from further down.